docs/_ext/scrapydocs.py

```diff
@@ -1,54 +1,62 @@
 # pylint: disable=import-error
+# ruff: noqa: PGH003
+from collections.abc import Sequence
 from operator import itemgetter
+from typing import Any
 
 from docutils import nodes
+from docutils.nodes import Element, General, Node, document
 from docutils.parsers.rst import Directive
-from docutils.parsers.rst.roles import set_classes
+from sphinx.application import Sphinx
 from sphinx.util.nodes import make_refnode
 
 
-class settingslist_node(nodes.General, nodes.Element):
+class settingslist_node(General, Element):
     pass
 
 
 class SettingsListDirective(Directive):
-    def run(self):
+    def run(self) -> Sequence[Node]:
         return [settingslist_node("")]
 
 
-def is_setting_index(node):
-    if node.tagname == "index" and node["entries"]:
+def is_setting_index(node: Node) -> bool:
+    if node.tagname == "index" and node["entries"]:  # type: ignore
         # index entries for setting directives look like:
         # [('pair', 'SETTING_NAME; setting', 'std:setting-SETTING_NAME', '')]
-        entry_type, info, refid = node["entries"][0][:3]
+        entry_type, info, refid = node["entries"][0][:3]  # type: ignore
         return entry_type == "pair" and info.endswith("; setting")
     return False
 
 
-def get_setting_target(node):
+def get_setting_target(node: Node) -> Node:
     # target nodes are placed next to the node in the doc tree
     return node.parent[node.parent.index(node) + 1]
 
 
-def get_setting_name_and_refid(node):
+def get_setting_name_and_refid(node: Node) -> tuple[Node, str]:
     """Extract setting name from directive index node"""
-    entry_type, info, refid = node["entries"][0][:3]
+    entry_type, info, refid = node["entries"][0][:3]  # type: ignore
     return info.replace("; setting", ""), refid
 
 
-def collect_scrapy_settings_refs(app, doctree):
+def collect_scrapy_settings_refs(app: Sphinx, doctree: document) -> None:
     env = app.builder.env
 
     if not hasattr(env, "scrapy_all_settings"):
-        env.scrapy_all_settings = []
+        env.scrapy_all_settings = []  # type: ignore
 
-    for node in doctree.traverse(is_setting_index):
-        targetnode = get_setting_target(node)
-        assert isinstance(targetnode, nodes.target), "Next node is not a target"
+    for node in doctree.findall(is_setting_index):
+        # This seems not needed anymore.
```

Comment:
Interesting, the page is indeed generated correctly AFAICS.

`def get_setting_target()` is also no longer needed then.

Reply:
May we remove the commented out code before merging?